src/Rating.js

```diff
@@ -1,8 +1,228 @@
 'use strict';
 
 import RatingBase from './Rating.soy';
+import { core } from 'metal';
 
 class Rating extends RatingBase {
+    /**
+     * @inheritDoc
+     */
+    constructor(opt_config) {
+        super(opt_config);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    attached() {
+        this.delegate('mouseover', 'a', this.handleMouseOverEvent.bind(this));
```

Comment:
Attaching these events here works well, but we now have the option of passing functions to the template and attaching them there as well, if you wish. For example, the `mouseout` listener could be passed to the `render` template, like this:

```
/**
 * @param handleMouseOutEvent
 */
{template .render}
  <div data-onmouseout="{$handleMouseOutEvent}">
  ...
{/template}

The ones for the `a` links can be added in a similar way, the only thing is that for now only the functions requested by the `render` template as a param are passed down, so you'd need to declare these there and pass them to the `rateItems` template directly instead of passing them through `data="all"`.

That said, it's fine to keep these events here, just letting you know about the features :)